aea/helpers/async_utils.py

```diff
@@ -477,3 +458,206 @@ def __init__(self, getters: List[Tuple[Callable[[Any], None], Callable]]):
         super().__init__(
             [self._make_getter(handler, getter) for handler, getter in getters]
         )
+
+
+ready_future: Future = Future()
+ready_future.set_result(None)
+
+
+class Runnable(ABC):
+    """Abstract Runnable class."""
+
+    def __init__(self, loop=None, threaded=False) -> None:
+        """
+        Init runnable.
+
+        :param loop: asyncio event loop to use.
+        :param threaded: bool. start in thread if True.
+
+        :return: None
+        """
+        if loop and threaded:
+            raise ValueError(
+                "You can not set a loop in threaded mode, cause own loop will be created"
+            )
+        self._loop = loop
+        self._threaded = threaded
+        self._task: Optional[asyncio.Task] = None
+        self._thread: Optional[Thread] = None
+        self._completed_event: Optional[asyncio.Event] = None
+        self._got_result = False
+        self._was_cancelled = False
+        self._is_running: bool = False
+
+    def start(self) -> bool:
+        """
+        Start runnable.
+
+        :return: bool started or not.
+        """
+        if self._task and not self._task.done():
+            logger.debug("already running")
+            return False
+        self._is_running = False
+        self._got_result = False
+        self._set_loop()
+        self._completed_event = asyncio.Event(loop=self._loop)
+        self._was_cancelled = False
+        self._set_task()
+
+        if self._threaded:
+            self._thread = Thread(
+                target=self._loop.run_until_complete, args=[self._task]
+            )
+            self._thread.start()
+
+        return True
+
+    def _set_loop(self) -> None:
+        """Select and set loop."""
+        if self._threaded:
+            self._loop = asyncio.new_event_loop()
+        else:
+            try:
+                self._loop = self._loop or asyncio.get_event_loop()
```

Comment:
if we agree with the comment above, we can just use `get_event_loop()`.

Reply:
no, for threaded mode new loop is required, also some checks for runtime is required as well, cause there is a case with new thread and no loop is available.